### ISSUE TITLE
Local notifications for Mentions on Android

### ISSUE THREAD

**Issue by hesterbruikman** [status-im/status-react#11213]
Blocked by #11117 

# Feature Issue
Add local notifications for mentions on Android

### User Story
- As a user I want to receive a local notification when someone mentions me
- As a user I want to enable/disable receiving notifications for mentions (add notification settings to Notification preferences)
- As a user I want to be able to enable/disable showing message sender and content in the notification
- As a user I want to see message sender and content in the notification by default 
- As a user I want to open the relevant chat where someone mentioned me when I tap on the mention notification
e.g. 
```
Teenage Mutant Turtle mentioned you
Hey @maciej
```
- As a user I want to tap on the local notification to go to the mention inside the chat where the mention was placed

### Acceptance Criteria
- When user is mentioned and has local notifications on Android enabled, mention notification appears when the app is closed
- User can disable `Show @mentions` in Notification preferences on Android
- When `Show @mentions is disabled, local notifications are no longer received
- User can `Hide message content and sender` in Notification preferences on Android
- When hide content and sender is disabled, notification shows `Someone mentioned you`
- When tapping on mention notification, app is opened, after log in, chat is opened where user was mentioned

### Notes
- This feature is blocked by #11117 However, if demand is high upon releasing mentions, effort needs to be reviewed to compare implementing #11117 along with this issue or implementing this issue standalone in the current Java implementation
- Future issue is to enable push notifications for mentions (and other) on Android

**Comment by guylouis** [comment]
@qoqobolo from your testing and remarks here https://github.com/status-im/status-react/pull/11653#issuecomment-774038002

would you say this one can be closed? 

cc @cammellos

**Comment by qoqobolo** [comment]
@guylouis 

in this PR, of all the acceptance criteria, the following were implemented:

- When user is mentioned and has local notifications on Android enabled, mention notification appears when the app is closed
- When tapping on mention notification, app is opened, after log in, chat is opened where user was mentioned

So closing is at your and @cammellos discretion.

**Comment by guylouis** [comment]
thanks @qoqobolo 

Let's have this one closed, and part of 1.12 release. 

I have created a new issue #11808 for the missing settings for mentions notification which are 
- User can disable Show @mentions in Notification preferences on Android
- When `Show @mentions is disabled, local notifications are no longer received
- User can Hide message content and sender in Notification preferences on Android
- When hide content and sender is disabled, notification shows Someone mentioned you
